plugins/generator-1.16.5/forge-1.16.5/templates/block.java.ftl

```diff
@@ -301,10 +301,14 @@ public class ${name}Block extends ${JavaModName}Elements.ModElement {
 		</#if>
 
 		<#if data.specialInformation?has_content || hasProcedure(data.specialInformation)>
-		@Override @OnlyIn(Dist.CLIENT) public void addInformation(ItemStack itemstack, IBlockReader reader, List<ITextComponent> list, ITooltipFlag flag) {
-			super.addInformation(itemstack, reader, list, flag);
+		@Override @OnlyIn(Dist.CLIENT) public void addInformation(ItemStack itemstack, IBlockReader blockReader, List<ITextComponent> list, ITooltipFlag flag) {
+			super.addInformation(itemstack, blockReader, list, flag);
 			<#if hasProcedure(data.specialInformation)>
-				World world = (World) reader;
+				ClientPlayerEntity entity = (ClientPlayerEntity) itemstack.getAttachedEntity();

```

Comment:
This isn't how you get the client player. Check how the glow condition procedure does it

Reply:
There's no glow condition

Reply:
Most elements have hasEffect

Reply:
Okay I changed it, how does it look now? I still have the issue where it tries to pass no argument idk why it's happening...